Competition2018/src/main/java/competition/subsystems/autonomous/AutonomousDecisionSystem.java

```diff
@@ -7,107 +7,156 @@
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
-import openrio.powerup.MatchData;
 import openrio.powerup.MatchData.GameFeature;
 import openrio.powerup.MatchData.OwnedSide;
 import xbot.common.command.BaseSubsystem;
 import xbot.common.math.ContiguousHeading;
 import xbot.common.math.FieldPose;
 import xbot.common.math.XYPair;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.XPropertyManager;
 
 @Singleton
 public class AutonomousDecisionSystem extends BaseSubsystem {
 
-    GameDataSource gameData;
+    final GameDataSource gameData;
+    final BooleanProperty robotOnRight;
     
     @Inject
-    public AutonomousDecisionSystem(GameDataSource gameData) {
+    public AutonomousDecisionSystem(GameDataSource gameData, XPropertyManager propMan) {
         this.gameData = gameData;
+        robotOnRight = propMan.createPersistentProperty(getPrefix() + "Robot On Right", true);
+    }
+    
+    public void setRobotPosition(boolean onRight) {
+        robotOnRight.set(onRight);
     }
     
     public Supplier<List<FieldPose>> getAutoPathToFeature(GameFeature feature) {
-        return () -> chooseBestPathToFeature(feature);
+        return () -> chooseBestPathToFeature(feature, robotOnRight.get());
     }
     
-    private List<FieldPose> chooseBestPathToFeature(GameFeature feature) {
-        OwnedSide targetSide = gameData.getOwnedSide(feature);
+    private List<FieldPose> mirrorPath(List<FieldPose> path) {
+        List<FieldPose> flippedPath = new ArrayList<FieldPose>();
+        
+        for (FieldPose pose : path) {
+            flippedPath.add(new FieldPose(
+                    new XYPair(-pose.getPoint().x, pose.getPoint().y),
+                    new ContiguousHeading(pose.getHeading().getValue())
+                )
+            );
+        }
         
+        return flippedPath;
+    }
+    
+    private List<FieldPose> chooseBestPathToFeature(GameFeature feature, boolean onRight) {
```

Comment:
The multi-layer switch statements with if/else inside goes on too long, but I think is the consequence of quick-fix situations elsewhere. 

We have slightly different behaviors on three dimensions:
Target object
Target side
Starting position

We could probably reduce that by collapsing the input variables somehow to
Target object
Are you close or far to the target plate